python-sdk/nuscenes/eval/predict/metrics.py

```diff
@@ -0,0 +1,436 @@
+        return self._aggregators
+
+    @property
+    def name(self):
+        return 'MinADEK'
+
+    @property
+    def shape(self):
+        return len(self.k_to_report)
+
+
+class MinFDEK(Metric):
+
+    def __init__(self, k_to_report, aggregators: List[Aggregator]):
+        super().__init__()
+        self.k_to_report = k_to_report
+        self._aggregators = aggregators
+
+    def __call__(self, ground_truth: np.ndarray, prediction: Prediction) -> np.ndarray:
+        ground_truth = stack_ground_truth(ground_truth, prediction.number_of_modes)
+        results = min_fde_k(prediction.prediction, ground_truth, prediction.probabilities)
+        return desired_number_of_modes(results, self.k_to_report)
+
+    def serialize(self) -> Dict[str, Any]:
+        return {'k_to_report': self.k_to_report,
+                'name': self.name,
+                'aggregators': [agg.serialize() for agg in self.aggregators]}
+
+    @property
+    def aggregators(self,) -> List[Aggregator]:
+        return self._aggregators
+
+    @property
+    def name(self):
+        return "MinFDEK"
+
+    @property
+    def shape(self):
+        return len(self.k_to_report)
+
+
+class MissRateTopK(Metric):
+
+    def __init__(self, k_to_report: List[int], aggregators: List[Aggregator],
+                 tolerance: float = 2.):
+        self.k_to_report = k_to_report
+        self._aggregators = aggregators
+        self.tolerance = tolerance
+
+    def __call__(self, ground_truth: np.ndarray, prediction: Prediction) -> np.ndarray:
+        ground_truth = stack_ground_truth(ground_truth, prediction.number_of_modes)
+        results = miss_rate_top_k(prediction.prediction, ground_truth,
+                                 prediction.probabilities, self.tolerance)
+        return desired_number_of_modes(results, self.k_to_report)
+
+    def serialize(self) -> Dict[str, Any]:
+        return {'k_to_report': self.k_to_report,
+                'name': 'MissRateTopK',
+                'aggregators': [agg.serialize() for agg in self.aggregators],
+                'tolerance': self.tolerance}
+
+    @property
+    def aggregators(self,) -> List[Aggregator]:
+        return self._aggregators
+
+    @property
+    def name(self):
+        return f"MissRateTopK_{self.tolerance}"
+
+    @property
+    def shape(self):
+        return len(self.k_to_report)
+
+
+class OffRoadRate(Metric):
+
+    def __init__(self, helper: PredictHelper, aggregators: List[Aggregator]):
+        self._aggregators = aggregators
+        self.helper = helper
+        self.drivable_area_polygons = self.load_drivable_area_masks(helper)
+        self.pixels_per_meter = 10
+
+    @staticmethod
+    def load_drivable_area_masks(helper: PredictHelper) -> Dict[str, np.ndarray]:
+        """
+        Loads the polygon representation of the drivable area for each map
+        :param helper: Instance of PredictHelper
+        :return: Mapping from map_name to drivable area polygon
```

Comment:
I recommend each docstring line ending in `.`.

Reply:
Ok, will implement across PR